forum/Sources/Tips.php

```diff
@@ -0,0 +1,81 @@
+<?php 
+
+if (!defined('SMF'))
+	die('Hacking attempt...');
+
+function Tips()
+{
+	global $context, $smcFunc, $user_info, $txt;
+
+	$context['page_title'] = $txt['tip_list_title'];
+	
+	loadPosts();
+
+	loadTemplate('Tips');
+}
+
+function loadPosts()
+{
+	global $memberContext, $context, $smcFunc, $user_info;
+	
+	$tippedPosts = array();
+	
+	// Find all tips in descending order
+	// TO-DO: custom-limit: pagination support, etc.
+	// Get most recent tipped posts
+	$query = 'SELECT id_message_tip, id_message, id_member, coins FROM {db_prefix}message_tips';
+
+	// Searching for posts tipped by a specific user?
+	if (!empty($_GET['tipper']) && !$context['user']['is_guest'])
+	{
+		// Find user id for searched user
+		$userSearch = $smcFunc['db_query']('', 'SELECT id_member from smf_members where real_name = \'' . mysql_escape_string(urldecode($_GET['tipper'])) . '\'');
+		$result = $smcFunc['db_fetch_assoc']($userSearch)['id_member'];
+		$query .= ' WHERE id_member = \'' . $result . '\'';
+	}
+
+	$query .= ' ORDER BY id_message_tip DESC LIMIT 15';
+
+	$tipsQuery = $smcFunc['db_query']('', $query);

```

Comment:
What is the first empty parameter for?

Reply:
SMF supports multiple different database systems. SMF lets you write all your queries in MySQL and if you pass an 'identifier' to this function you can use other databases like SQLite.

So if you put "substring" as the first parameter of that function, SUBSTRING would get replaced with SUBSTR.